src/Sylius/Bundle/CoreBundle/Handler/CompositeLocaleChangeHandler.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Handler;
+
+use Sylius\Component\Core\Exception\HandleException;
+use Sylius\Component\Core\Locale\Handler\LocaleChangeHandlerInterface;
+use Zend\Stdlib\PriorityQueue;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class CompositeLocaleChangeHandler implements LocaleChangeHandlerInterface
+{
+    /**
+     * @var PriorityQueue|LocaleChangeHandlerInterface[]
+     */
+    private $handlers;
+
+    public function __construct()
+    {
+        $this->handlers = new PriorityQueue();
+    }
+
+    /**
+     * @param LocaleChangeHandlerInterface $localeChangeHandler
+     * @param int $priority
+     */
+    public function addHandler(LocaleChangeHandlerInterface $localeChangeHandler, $priority = 0)
+    {
+        $this->handlers->insert($localeChangeHandler, $priority);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($code)
+    {
+        if ($this->handlers->isEmpty()) {
+            throw new HandleException(self::class, 'There is no defined handlers.');
+        }
+
+        foreach ($this->handlers as $handler) {
+            try {
+                $handler->handle($code);
+            } catch (HandleException $exception) {
+                throw new HandleException(
```

Comment:
Do we need this? Looks strange to catch exception and then wrap it with the same type of exception and rethrow. We have stack trace, so I think this is redundant and not readable.